runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6393,6 +6418,148 @@ TR_J9ByteCodeIlGenerator::genFlattenableWithFieldWithHelper(uint16_t fieldCpInde
+
+   newName[newStringLen] = '\0';
+   return newName;
+   }
+
+
+void
+TR_J9ByteCodeIlGenerator::genFlattenableWithField(uint16_t fieldCpIndex, TR_OpaqueClassBlock * valueClass)
+   {
+   /* An example on what the tree with flattened fields would look like
+    *
+    * value FlattenedLine2D {
+    *   flattened Point2D st;
+    *   flattened Point2D en;
+    * }
+    * /--- trees inserted ------------------------
+    * n68n     (  0)  NULLCHK on n63n
+    * n67n     (  2)    iloadi  FlattenedLine2D.st.x
+    * n63n     (  3)      ==>newvalue (Identityless sharedMemory )
+    * n70n     (  0)  NULLCHK on n63n
+    * n69n     (  2)    iloadi  FlattenedLine2D.st.y I
+    * n63n     (  3)      ==>newvalue (Identityless sharedMemory )
+    * n72n     (  0)  NULLCHK on n65n
+    * n71n     (  2)    iloadi  Point2D.x I
+    * n65n     (  2)      aload  <parm 1 P>
+    * n74n     (  0)  NULLCHK on n65n
+    * n73n     (  2)    iloadi  Point2D.y I
+    * n65n     (  2)      ==>aload
+    * n76n     (  0)  treetop
+    * n75n     (  1)    newvalue  jitNewValue[#100  helper Method]
+    * n66n     (  1)      loadaddr  FlattenedLine2D
+    * n67n     (  2)      ==>iloadi
+    * n69n     (  2)      ==>iloadi
+    * n71n     (  2)      ==>iloadi
+    * n73n     (  2)      ==>iloadi
+    * /--- stack after ------------------------
+    * @0 n75n     (  1)  ==>newvalue (Identityless sharedMemory )
+    * ============================================================
+    */
+   TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
+
+   if (isFieldResolved(comp(), owningMethod, fieldCpIndex, false))
+      {
+      TR::Node *newFieldValue = pop();
+      TR::Node *originalObject = pop();
+
+      int32_t prefixLen = 0;
+      char * fieldNamePrefix = getTopLevelPrefixForFlattenedFields(owningMethod, fieldCpIndex, prefixLen, comp()->trMemory()->currentStackRegion());
+
+      int len;
+      const char * fieldClassChars = owningMethod->fieldSignatureChars(fieldCpIndex, len);
+      TR_OpaqueClassBlock * fieldClass = fej9()->getClassFromSignature(fieldClassChars, len, owningMethod);
+
+      loadClassObject(valueClass);
+
+      const TR::TypeLayout *typeLayout = comp()->typeLayout(valueClass);
+      size_t fieldCount = typeLayout->count();
+
+      TR_OpaqueClassBlock * containingClass = owningMethod->definingClassFromCPFieldRef(comp(), fieldCpIndex, _methodSymbol->isStatic());
+
+      for (size_t idx = 0; idx < fieldCount; idx++)
+      {
+      const TR::TypeLayoutEntry &fieldEntry = typeLayout->entry(idx);
+      if (!strncmp(fieldNamePrefix, fieldEntry._fieldname, prefixLen))
+         {
+         char * fieldNameRemovedTopLevelPrefix = removeTopLevelPrefixForFlattenedFields(fieldEntry._fieldname, prefixLen,
+                                                                                        comp()->trMemory()->currentStackRegion());
```

Comment:
Does this actually need to create a copy of the name with the prefix removed or could it simply be written as follows?

```
char *fieldNameRemovedTopLevelPrefix = fieldEntry._fieldname + prefixLen;
```

Reply:
Ah, it is 👍. No need to do all those manipulation in `removeTopLevelPrefixForFlattenedFields`